src/neml2/models/solid_mechanics/KocksMeckingActivationEnergy.cxx

```diff
@@ -0,0 +1,92 @@
+// Copyright 2023, UChicago Argonne, LLC
+// All Rights Reserved
+// Software Name: NEML2 -- the New Engineering material Model Library, version 2
+// By: Argonne National Laboratory
+// OPEN SOURCE LICENSE (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include "neml2/models/solid_mechanics/KocksMeckingActivationEnergy.h"
+
+namespace neml2
+{
+register_NEML2_object(KocksMeckingActivationEnergy);
+
+OptionSet
+KocksMeckingActivationEnergy::expected_options()
+{
+  OptionSet options = Model::expected_options();
+  options.doc() =
+      "Calculates the Kocks-Mecking normalized activation as \\f$g = \\frac{kT}{\\mu "
+      "b^3} \\log \\frac{\\dot{\\varepsilon}_0}{\\dot{\\varepsilon}} \\f$ with \\f$ "
+      "\\mu \\f$ the shear modulus, \\f$ k \\f$ the Boltzmann constant, \\f$ T \\f$ the absolute "
+      "temperature, \\f$ b \\f$ the Burgers vector length, \\f$ \\dot{\\varepsilon}_0 \\f$ a "
+      "reference strain rate, and \\f$ \\dot{\\varepsilon} \\f$ the current strain rate.";
+
+  options.set<CrossRef<Scalar>>("shear_modulus");
+  options.set("shear_modulus").doc() = "The shear modulus";
```

Comment:
Just out of curiosity, how does this work with anisotropic material?

Reply:
Actually NEML handles this case.  For this (macroscale) model it calculates the average shear modulus directly from the elasticity objects.  At some point we should do that in NEML2.

For similar crystal models it actually uses the correct shear modulus on the slip plane.  Again the NEML elasticity objects do that.